lib/command.js

```diff
@@ -44,19 +20,21 @@ if (program.type != null &&
   errors.push ('Invalid type `' + program.type + "'");
 }
 if (errors.length > 0) {
-  process.stderr.write (formatErrors (errors));
+  process.stderr.write (common.formatErrors (errors));
   process.exit (1);
 }
 
-process.exit (program.args.reduce (function(status, path) {
-  var results;
-  try {
-    results = doctest (path, program);
-  } catch (err) {
-    process.stderr.write (formatErrors ([err.message]));
-    process.exit (1);
-  }
-  return results.reduce (function(status, tuple) {
-    return tuple[0] ? status : 1;
-  }, status);
-}, 0));
+Promise.all (program.args.map (function(path) {
+  return doctest (path, program).then (function(results) {
+    return results.reduce (function(status, tuple) {
+      // Note: This will cause a non-zero-exit to go back to zero.
+      // I think it's a bug, but it was already there.
+      return tuple[0] ? status : 1;
+    }, 0);
```

Comment:
`status` is the accumulator. It's `0` initially and can go from `0` to `1`, but can never go from `1` back to `0`.

Reply:
Yeah. But with the code being like this, if I'm not mistaken, the last iteration always determines the output code, independent of what happened in the other iterations.

Reply:
Ah, never mind. I am actually mistaken.